### DISCUSSION TITLE
ESP32-S3  5V IN verursacht Reboot

### DISCUSSION THREAD

**Discussion by Tpaydt** [hoylabs/OpenDTU-OnBattery#1177] [Q&A] [answered]
Ich bin auf ein ESP32-S3 DevKitC-1 umgestiegen wegen dem neuen 2024.08.18 Release, alles läuft wunderbar, dickes Respekt an die Macher.
Wie beim Vorgänger ESP32 wollte ich die Stromversorgung über die 5V IN Pins realisieren, leider booten dann der ESP32-S3 nicht mehr.
Über die beiden USB Ports läuft alles perfekt, Netzteil ist ein Handylader mit 5V und 1,6A.
Denke dass es sich um ein Hardwareproblem handelt, deshalb die Frage ob jemand ein ESP32-S3 auch über 5V IN betreibt?

Hier der PinOut von meinem:
![ESP32-S3 PINOUT](https://github.com/user-attachments/assets/27cacc17-6ee8-42d2-8e84-39a12389ed1d)

**Comment by schlimmchen** [comment]
Ich hab auch so ein Board zum Spielen bestellt, hab es aber noch nicht angerührt. Mein erster Tipp wäre dieser hier: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/ESP32-Versions-and-Memory#important-info-for-those-intending-to-use-the-5v-as-output Hast du so eine Lötbrücke? Ist die offen oder kurzgeschlossen? Soweit ich das interpretiere (ich habs nirgends sicher nachgelesen oder gar ausprobiert) muss diese Brücke geschlossen sein, wenn man vom USB-Port 5V an den 5V Pin bringen möchte und offen sein, wenn man 5V Spannungsversorgung über diesen Pin an das Board machen möchte.

**Comment by Tpaydt** [comment]
hey, ja das hab ich auch schon gelesen aber meine Brücke ist offen, habe diese auch auf Durchgang geprüft.
Falls du mal Zeit findest und bei deinem Board es mal testen könntest, wäre ich dankbar.

ich versuch mal einen LOG zu machen ohne mein USB Port am Notebook zu schroten

**Comment by Solarteur** [comment]
Ich habe dieses Board in der Variante "ESP32-S3 N16R8":
https://de.aliexpress.com/item/1005004617322170.html?spm=a2g0o.order_list.order_list_main.97.21ef5c5fejPeDP&gatewayAdapt=glo2deu

Die oben schon zitierte Beschreibung 
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/ESP32-Versions-and-Memory#important-info-for-those-intending-to-use-the-5v-as-output
ist für meinen Fall zutreffend: Die 5V-Spannungsversorgung für das Board ist ohne Probleme und ohne weitere Maßnahmen über den Pin "5V in" möglich. 
Soll Pin "5V in" als "5V out" genutzt werden, überbrückt die Lötbrücke "IN-OUT" nach meinen Messungen eine Diode.

**Comment by SW-Niko** [comment]
Ich benutze auch den 5V Eingang auf dem ESP32-S3 Kit. 
Zeitweise wird das Board parallel über die USB 5V versorgt. Durch die Entkopplung über die Dioden funktioniert das problemlos auf meinem Kit. 
Anscheinend gibt es verschiedene Varianten vom ESP32-S3 DevKitC-1. Ich habe die Version 1.1 und da ist der 5V Pin direkt und ohne Diode oder Brücke mit den internen 5V verbunden.


![grafik](https://github.com/user-attachments/assets/7d606375-5d81-4111-ac08-55ba328e10a2)


![grafik](https://github.com/user-attachments/assets/50897153-c259-4f5f-b3b0-fbc58b25ba8b)

**Comment by schlimmchen** [comment]
> ich versuch mal einen LOG zu machen ohne mein USB Port am Notebook zu schroten

Moment... Das Ding kriegt zwar Strom und tut irgendetwas, aber es startet immer wieder neu?! Das würde ja bedeuten, dein Netzteil taugt nichts :man_shrugging: Also dass es fürs vorherige Boards ausreichend war, aber fürs neue nicht mehr.

**Comment by Manos1966** [comment, marked as answer]
> Moment... Das Ding kriegt zwar Strom und tut irgendetwas, aber es startet immer wieder neu?! Das würde ja bedeuten, dein Netzteil taugt nichts 🤷‍♂️ Also dass es fürs vorherige Boards ausreichend war, aber fürs neue nicht mehr.

Oder, @Tpaydt , deine Pin Konfiguration ist falsch.
- alle angeschlossene HW entfernen
- leere Pin Config laden (falls nicht moeglich, neu flashen)
und dann testen ob der Boot Vorgang funktioniert hat.

und FALLS du GPIO 1 & GPIO 2 benutzt hast, dann ist dies zu 90% der Grund deiner Misere.... 

UPDATE: Entschuldigung, ich habe deine erste Post nicht richtig verstanden. Du hast bereits gesagt, ueber USB Typ-C Stromversorgung hast du keine Probleme.
Dann, nur "- alle angeschlossene HW entfernen" und testen (die PIN Konfiguration kann keinen Einfluss haben).

**Reply by woollfi** [reply]
Warum darf man gpio 1 und 2 nicht nehmen, welche darf man verwenden und welche sollte man meiden?

**Reply by schlimmchen** [reply]
Das kommt auf deinen ESP32 an, also ob es ein ESP32 oder ein ESP32-S3 oder sonstetwas ist. GPIO0 ist immer zum starten im bootloader, den solltest du immer vermeiden. Dann sind am ESP32 jene zwei zu vermeiden, die zum UART0 gehören, da kommen nämlich die seriellen Nachrichten raus. Beim ESP32 sind das GPIO1 und GPIO3. Beim ESP32-S3 sind das GPIO43 und GPIO44. Am ESP32-S3 sind dann noch 35, 36 und 37 zu meiden, siehe auch https://opendtu-onbattery.net/hardware/esp32s3_devkit/. Deine Frage ist insgesamt berechtigt und das sollte man mal aufschreiben in die doku...

**Reply by Snusme83** [reply]
@schlimmchen : Bin gerade über diesen Beitrag gestolpert. Könnte GPIO 35-37 dafür verantwortlich sein, dass meine Verbindung zwischen NRF24-Modul und Hoymiles HM-1500 so bescheiden ist? Ich verwende einen ESP32-S3.

**Comment by Tpaydt** [comment]
der Fehler ist gefunden, es lag tatsächlich am falschen Pin Config Displaytyp. Ich hab ein OLED 2,42 SSD1309 angeschlossen, welches ja nach dem umstieg auf ESP32-S3 gleich funktionier hat mit der USB Verbindung und auch über 5V IN. Deswegen hatte ich nicht mehr danach geschaut und hatte vergessen das ich damals von Hand den Displaytyp auf "4" ändern muss. 
Jetzt läuft es mit 5V IN.  Danke an alle.

**Reply by lopo2005ch** [reply]
Hi  Tpaydt,
ich beiss mir gerade die Zähne aus. 
Das Board von reichelt ESP32-S3-WROOM-1-N8, das  DEBO OLED 2.42, OLED ist auf i2c mit BS1 und BS2 umgelötet.
Ich habe:
        "display": {
            "type": 4,
            "data": 5,
            "clk": 17
        }
Bekomme damit aber kein Lebenszeichen des Display.
Noch ist zwar auch der nrf24 nicht angeschlossen, dass Display müsste doch trotzdem gehen, oder?
Bei dem  nrf24 stehe ich vermutlich vor dem gleichen Problem: wohin gehört welcher Pin.
Könntest Du mal Deine *.json posten?
und einen Kommentar mit der entsprechenden Belegung?
Gruss Lopo

**Reply by Tpaydt** [reply]
Hi

Du musst den DC pin zusätzlich auf Masse legen.

The DC pin can be used to change the I2C address: Set Low, the 7-bit I2C address is 0x3C; Set High, the I2C address is 0x3D.

**Reply by frank-tanne** [reply]
Display Reset muss noch auf RST beim ESP32

**Reply by frank-tanne** [reply]
Und ja ich betreibe das ESP32 s3 Port über den 5v Pin.

**Reply by lopo2005ch** [reply]
Hi Frank, 
Display Reset auf RST löte ich morgen um.

Hi Tpaydt, 
das mit dem DC Pin verstehe ich nicht.  DC habe ich / sehe ich nicht auf dem OLED Board.

LG Lopo

**Reply by Tpaydt** [reply]
ich habe dieses oled display
https://www.waveshare.com/2.42inch-oled-module.htm

**Comment by woollfi** [comment]
hast du auch das display aktiviert, wenn der inverter nicht erreichbar ist?

liebe grüße

Am Mo., 21. Okt. 2024 um 19:11 Uhr schrieb lopo2005ch <
***@***.***>:

**Reply by lopo2005ch** [reply]
Guten Morgen woollfi,
 jo hab ich.
Eine DTU o B mit ESP32-D0WD und 0.9" OLED läuft seit vielen Monaten einwandfrei, weil das Display mir zu klein ist wollte ich ein grösseres und weil es mit generic_esp32 immer mal Update Probleme gab wollte ich den ESP32-S3 nehmen.
 LG Lopo

**Comment by woollfi** [comment]
Hast du vielleicht böse Pins belegt?

Von meinem iPhone gesendet

**Comment by lopo2005ch** [comment]
![Verdratung](https://github.com/user-attachments/assets/93ad09ac-5924-4941-8eba-b1f580a98c0a)
Von Joy-IT hatte mir jemand diese Beschaltung des Display aufgezeichnet, nicht in Verbindung mit dem ESP32-S3.
Vermutlich ist dort irgendwo der Fehler.
Vielen Dank an die Helfer!

**Reply by frank-tanne** [reply]
Soviel mir bekannt ist braucht das Display nur 3,3v . Sonst hast du nicht lange freute. Das Blaue Kabel muss an RST und nicht an D2

**Reply by lopo2005ch** [reply]
Hi Frank, 
ok, das mit dem D2 ist halt die Aussage bzw Skizze des Technikers von Joy-IT.
Aber egal, ich baue mal um.

**Reply by Sorbat** [reply]
Welche Funktion hat Joy-IT hierbei?

**Comment by jab42** [comment]
Welche Belegung brauche ich bei diesem Board denn für nrf und cmt? Sobald ich das Profil für beides aktiviere, bootet er nicht mehr und ich muss neu flashen. Nur mit nrf klappt es.
```
        "name": "NRF24 + CMT2300A",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "cmt": {
            "clk": 12,
            "sdio": 14,
            "cs": 27,
            "fcs": 26,
            "gpio2": -1,
            "gpio3": -1
        }
```
Der Fehler: [esp32-hal-gpio.c:102] __pinMode(): Invalid pin selected

**Reply by woollfi** [reply]
"nrf24": {
            "miso": 14,
            "mosi": 9,
            "clk": 10,
            "irq": 13,
            "en": 12,
            "cs": 11
        }, 
	"cmt": {
            "clk": 6,
            "cs": 4,
            "fcs": 5,
            "sdio": 7,
            "gpio3": 21
        }

**Reply by jab42** [reply]
super, werd ich versuchen. danke.
